src/components/Tree/Tree.tsx

```diff
@@ -122,28 +121,24 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
 
         case 'REGISTER_NODE_CHILDREN':
             {
-                const { id, children } = action.payload;
+                const { id: parentId, children } = action.payload;
 
-                const nodeIndex = findIndexById(draft.nodes, id);
+                const parentNodeIndex = findIndexById(draft.nodes, parentId);
 
-                if (nodeIndex === -1) {
-                    console.error(`Element with ID "${id}" not found.`);
+                if (parentNodeIndex === -1) {
+                    console.error(`Element with ID "${parentId}" not found.`);
                     return;
                 }
 
-                const currentChildrenIds = getNodeChildrenIds(draft.nodes, id);
+                const currentChildrenIds = getNodeChildrenIds(draft.nodes, parentId);
 
-                const newChildrenIds = children.map((node) => node.props.id as string);
+                const newChildrenIds = children.map((node) => node.props.id);
```

Comment:
Fyi this will have type any[] if we remove 'as string'.

Reply:
Not anymore
![Screenshot 2023-05-08 at 09 19 47](https://user-images.githubusercontent.com/6846950/236760587-9f6e6e78-6dc5-4fa9-be3f-90c17dfc66cf.png)